### ISSUE TITLE
ExistsResponse.IsValid is false for not-existing index

### ISSUE THREAD

**Issue by maxsivkov** [elastic/elasticsearch-net#4684]
**NEST/Elasticsearch.Net version**:
7.6.1
**Elasticsearch version**:
7.6.0
**Description of the problem including expected versus actual behavior**:
Nest.ElasticClient.Indices.Exists and Nest.ElasticClient.Indices.ExistsAsync incorrectly set ExistsResponse.IsValid property in the case index does not exists

**Steps to reproduce**:
 1. create client cli
 2. execute response = cli.Indices.Exists("notexistet")

**Expected behavior**
response.IsValid should be true
response.Exists should be false

**Explanation:**
ExistsResponse is derived from ResponseBase
ResponseBase class has
```C#
public virtual bool IsValid 
{
  get
  {
    var statusCode = ApiCall?.HttpStatusCode;
    if (statusCode == 404) return false;
    return (ApiCall?.Success ?? false) && ServerError == null;
  }
}
```
I think it is totally valid for all cases except ExistsResponse :)

**Solution:**
Override *IsValid*  in *ExistsResponse*  class removing `if (statusCode == 404) return false;` check
```C#
public override bool IsValid => (ApiCall?.Success ?? false) && ServerError == null;
```

ExistsResponse.cs should be like this:
```C#
using System.Runtime.Serialization;

namespace Nest
{
	[DataContract]
	public class ExistsResponse : ResponseBase
	{
		public bool Exists => ApiCall != null && ApiCall.Success && ApiCall.HttpStatusCode == 200;

		public override bool IsValid => (ApiCall?.Success ?? false) && ServerError == null;
	}
}

```

**Comment by russcam** [comment]
Hi @maxsivkov 

The implementation for how all responses are handled and deserialized was revisited in 7.x and the implementation for `ExistsResponse` also revisited.

It's been discussed many times over the years, on issues and internally, about what the semantics should be for `IsValid` and exists responses:

- #3707
- #3873
- #1621

The decision at this time, implemented in #3873 is that a 404 is marked as `.IsValid = false`, so that it signals to the consumer that `.Exists` and status codes should also be checked. This change is listed in the [6.x -> 7.x breaking changes docs](https://www.elastic.co/guide/en/elasticsearch/client/net-api/current/nest-breaking-changes.html#_high_level_client).

I'm going to close this issue as I don't think we'd want to change this behaviour going forward.

**Comment by StingyJack** [comment]
It keeps getting discussed and brought up because this is another part of this API that doesn't follow a well known or logical convention, and it has not been fixed despite many requests to do so. 

A property like `IsValid` (in all its forms - `Ok`, `IsOk`, `ProcessedOk`, `Success`, etc) with a value of `true` on a response payload should be informing callers that the request was processed without issue and that any other property value on the response has a complete and legitimate value in it.

Does the index exist? These are the expected permutations.
 - IsValid = False   (could not determine if the index exists, probably retry but fail the calling program at this point)
 - IsValid = True, Exists = False (determined it does not exist)
 - IsValid = True, Exists = True (determined if does exist)

I'm sure it has a proper name, but this kind of double boolean is not just a .net or even PC software convention, its even used in things like PLC's with ladder logic and physical device connections.
